### PR TITLE
Enable CI for external PR

### PR DESCRIPTION
**Pull request by Haichao-Zhang**
Enable CI for PRs created from a forked repo.

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -1,12 +1,19 @@
...
+on: [push, pull_request_target]
 
 jobs:
   build:
     runs-on: ubuntu-18.04
     container:
       image: horizonrobotics/alf:0.0.3-pytorch
+    # always run on push events, but only run on pull_request_target events
+    # when the pull request are from fork repositories; for pull requests
+    # within the same repository, the pull event is sufficient
```
Comment (emailweixu) [inline .github/workflows/test.yml]: push event is sufficient

Comment (Haichao-Zhang) [reply .github/workflows/test.yml]: fixed

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -1,12 +1,19 @@
...
     runs-on: ubuntu-18.04
     container:
       image: horizonrobotics/alf:0.0.3-pytorch
+    # always run on push events, but only run on pull_request_target events
+    # when the pull requests are from fork repositories; for pull requests
+    # within the same repository, the push event is sufficient
+    if: >
+      github.event_name == 'push' ||
+      github.event_name == 'pull_request_target' &&
+      github.event.pull_request.head.repo.full_name != github.repository
```
Comment (emailweixu) [inline .github/workflows/test.yml]: use parenthesis to group operations to make it less confusing

Comment (Haichao-Zhang) [reply .github/workflows/test.yml]: done

**Review by emailweixu — APPROVED** [review]

**Pull request merged**